docs/proposal/20211207-delegated-storage.md

```diff
@@ -0,0 +1,1186 @@
+
+    > **NOTE:** See also the [Rejected ideas](#rejected-ideas) section to learn why a generic validation idea was rejected.
+
+1. To install new storage backend, System Administrator has two options:
+
+    - use Capact Actions (e.g. `cap.interface.capactio.capact.hub.storage.helm-release.install`).
+    - Register a storage backend by creating such TypeInstance.
+
+    Regardless the option, at the end there is one TypeInstance produced:
+      
+      ```yaml
+      id: 3ef2e4ac-9070-4093-a3ce-142139fd4a16
+      typeRef:
+        path: cap.type.helm.storage
+        revision: 0.1.0
+      latestResourceVersion:
+        metadata:
+          alias: helm-storage # new field, more user-friendly description of such TypeInstance
+          attributes:
+          - path: "cap.core.attribute.hub.storage.backend" # related to GraphQL implementation
+            revision: 0.1.0
+        value:
+          url: "helm-release.default:50051"
+          acceptValue: false
+          additionalParametersSchema: |-
+            {
+              "$id": "#/properties/additionalParametersSchema",
+              "type": "object",
+              "required": [
+                "name",
+                "namespace"
+              ],
+              "properties": {
+                "name": {
+                  "$id": "#/properties/additionalParametersSchema/properties/name",
+                  "type": "string"
+                },
+                "namespace": {
+                  "$id": "#/properties/additionalParametersSchema/properties/namespace",
+                  "type": "string"
+                }
+              },
+              "additionalProperties": false
+            }
+      backend:
+        id: "a36ed738-dfe7-45ec-acd1-8e44e8db893b" # new immutable property - contains TypeInstance ID
+          # if not provided during TypeInstance creation, fallback to default one (get TypeInstance with proper Attribute and write its ID in this property)
+      ```
+
+1. In fresh Capact installation, there is one TypeInstance already preregistered:
+
+    ```yaml
+    id: a36ed738-dfe7-45ec-acd1-8e44e8db893b
+    typeRef:
+        path: cap.core.type.hub.storage.postgresql
+        revision: 0.1.0
+    latestResourceVersion:
+      metadata:
+        alias: capact-postgresql
+        attributes:
+        - path: cap.core.attribute.hub.storage.backend
+          revision: 0.1.0
+      value:
+        url: "storagebackend-postgresql.capact-system:50051"
+        acceptValue: true
+        additionalParametersSchema: null
+    backend: 
+      abstract: true # Special keyword which specifies the built-in storage option which stores already all other metadata. Effectively, it would be the same database as the PostgreSQL accessed via Capact storage backend service (`storagebackend-handlers.capact-system:50051`), but accessed directly.
```

Comment:
> EDIT: after reading the whole doc still cannot see it, but based on other sentences, my assumption is that Local Hub has PostgreSQL backend to store all data, and also registers that backend as the default for others, am I right?

Yes, it's more or less the same as how `capact-postgresql`, but avoiding the storage backend service between Hub and PostgreSQL database. It would be a different PSQL table, that's for sure.

And yes, we can have multiple such backends, see the Go template backend, it will also be `abstract`.

Reply:
Discussed f2f